### DISCUSSION TITLE
Do we have any plans to support all the raw apis?

### DISCUSSION THREAD

**Discussion by yueyinqiu** [magiccodingman/Magic.IndexedDb#19] [Ideas]
Actually I tried to do this 2 years ago, but failed: https://github.com/Nololiyt/Nololiyt.IndexedDbSharp.Raw.git

I think the reason why it failed is that I chose to wrap all the js stuffs. Instead, we should just provide equivalent apis in C#, and not to expose those js events/promises.

So do we have any plan to support those raw apis? I mean, to have apis in C# that are similar to [IndexedDB_API](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API).

This may break many things. We would have classes for databases, objectStores, transactions, keys... Dexie.js may not satisfy the needs as well.

If we do, I may try that again now.

**Comment by magiccodingman** [comment]
@yueyinqiu Ever have any success with it? I looked into something similar many years ago, but it didn't exist whenever I looked into it.

**Reply by yueyinqiu** [reply]
Haha problems exist at the beginning: How to keep the feature of [IDBOpenDBRequest.upgradeneeded](https://developer.mozilla.org/en-US/docs/Web/API/IDBOpenDBRequest/upgradeneeded_event)? We could access `db` in that scope. But maybe not all the features? And that will also [block](https://developer.mozilla.org/en-US/docs/Web/API/IDBOpenDBRequest/blocked_event) other connections.

I think I'm a bit confused that at that time, so I just stopped here.

Hmm actually... maybe that's not a great problem. What we should do is just clear our mind and figure out the correct behavior, or just skip those features for now.

I would continue to try this later. I think raw api support is really useful.

**Comment by yueyinqiu** [comment]
Hmm... still find it tricky in some details... it seems  impossible to avoid event callback. Perhaps we should start from the experience of C# itself without fully implementing the raw api.
